client/rpc/Cargo.toml

```diff
@@ -37,6 +37,7 @@ sp-rpc = { version = "6.0.0", path = "../../primitives/rpc" }
 sp-runtime = { version = "6.0.0", path = "../../primitives/runtime" }
 sp-session = { version = "4.0.0-dev", path = "../../primitives/session" }
 sp-version = { version = "5.0.0", path = "../../primitives/version" }
+sc-network = { version = "0.10.0-dev", path = "../network" }
```

Comment:
I would say this is an overkill to add sc-network for only one struct.

Reply:
Noob question, but what's the best way to address this then?

Reply:
I think to move it to sc-core

Reply:
I would actually suggest to copy-paste this struct.
It only makes sense for two things to share the same code when these two things are the same, but this is not the case here. There's no reason to hard-lock "multiaddr with PeerId" in the RPC layer as always being the same thing as "multiaddr with PeerId" in the CLI.

Reply:
I've changed the implementation to accept Multiaddr instead, since most of the functions in NetworkService accept Multiaddr, and I would have had to import libp2p in the RPC layer anyway if I wanted to copy and paste the struct.